examples/source/interactivity-dynamic-content/Dynamic_Accordion.html

```diff
@@ -0,0 +1,171 @@
+    <style amp-custom="">
+      .show {
+        display: block;
+      }
+      .hide {
+        display: none;
+      }
+    </style>
+  </head>
+  <body>
+      <h2>
+        Select which climate you prefer
+      </h2>
+      <!-- ## Implementation -->
+      <!--
+          The `input` section sets `amp-state` as `{climate: 1}` if 'Hot' is selected and `{climate: 2}` if 'Cold' is selected.
+          The `amp-accordion` is hidden by default, but displayed if climate != 0 (so either hot or cold must be selected).
+
+          The `src` for climateCountries is dynamic. If `{climate == 1}` then `climateCountries` data is pulled from the hot.json file. Else, if `{climate == 2}` then climateCountries data is pulled from the cold.json file
+
+          Each `amp-accordion` section is a different continent, therefore the data `climateCountries` also must be filtered by continent. The filtering is done in the `[src]` section in the `amp-list`.
+          Additionally, height is dynamically set based on number of items in filtered state
+        -->
+          <input
+            type="radio"
+            id="f-option1"
+            name="selector1"
+            role="button"
+            tabindex="0"
+            on="change:AMP.setState({climate:(event.checked) ? 1 : 0})"
+          />
+          <label for="f-option1">Warm</label>
+          <input
+            type="radio"
+            id="f-option2"
+            name="selector1"
+            on="change:AMP.setState({climate: (event.checked) ? 2 : 0})"
+            role="button"
+            tabindex="0"
+          />
+          <label for="f-option2">Cold</label>
+      <div id="accordion" [class]="(climate != 0) ? 'show':'hide'" class="hide">
+        <p>The accordion below lists countries, organized by continent, that match the climate you chose. </p>
+        <amp-state
+          id="climateCountries"
+          [src]="climate == 1 ? '/static/samples/json/hot.json': climate == 2 ? '/static/samples/json/cold.json': ''"
+        >
+        </amp-state>
+        <amp-accordion>
+          <section>
+            <h4>
+              Asia
+            </h4>
+              <amp-list
+                width="auto"
+                layout="fixed-height"
+                binding="refresh"
+                [src]="climateCountries.countries.filter(country => country.continent == 'Asia')"
+                [height]="climateCountries.countries.filter(country => country.continent == 'Asia').length * 35"
```

Comment:
Use https://amp.dev/documentation/components/amp-list/#[is-layout-container]-(optional) instead of binding the height: `[is-layout-container]=true`.

Reply:
updated